src/apis/generations.ts

```diff
@@ -41,25 +41,74 @@ export interface PromptsCreateNonStreaming extends PromptBodyBase {
 	stream?: false;
 }
 
+export interface Functions {
+	name?: Optional<string>;
+	description?: Optional<string>;
+	parameters?: Optional<object>;
+}
+
+export interface Tool{
+	function?: Optional<Functions>;
+	type?: Optional<string>;
+}
+
+export interface Messages {
+	content?: Optional<string>;
+	role?: Optional<string>;
+}
+  
 export type PromptsCreateParams = PromptsCreateNonStreaming | PromptsCreateStreaming
 
 type PromptsResponse = Record<string, any> & APIResponseType;
 
+type Optional<T> = T | null | undefined;
+
+type PromptRenderResponse = {
+	success: boolean;
+	data: {
+		messages?: Optional<Messages[]>;
```

Comment:
Is Optional required here? All the params are already using `?:`.

Reply:
Same for other newly added types as well

Reply:
Yes, because it could be either matching completion or chat completion 
Hence keeping them all optional